pkg/config/collector/collector.go

```diff
@@ -201,41 +201,24 @@ func listFiles(dir string) ([]string, error) {
 
 // ParseCmdLine reads options from the kernel cmdline and returns the equivalent
 // Config.
-func ParseCmdLine(file string) (*Config, error) {
-	result := &Config{}
-
-	if file == "" {
-		file = "/proc/cmdline"
-	}
-	dat, err := os.ReadFile(file)
+func ParseCmdLine(file string, filter func(d []byte) ([]byte, error)) (*Config, error) {
+	result := Config{}
+	dotToYAML, err := machine.DotToYAML(file)
```

Comment:
not related to the change at hand, but `DotToYAML` doesn't sound something belonging to "machine" (SDK)